tempodb/blocklist/poller.go

```diff
@@ -226,69 +256,141 @@ func (p *Poller) pollTenantAndCreateIndex(ctx context.Context, tenantID string)
 	return blocklist, compactedBlocklist, nil
 }
 
-func (p *Poller) pollTenantBlocks(ctx context.Context, tenantID string) ([]*backend.BlockMeta, []*backend.CompactedBlockMeta, error) {
-	blockIDs, err := p.reader.Blocks(ctx, tenantID)
+func (p *Poller) pollTenantBlocks(
+	ctx context.Context,
+	tenantID string,
+	previous *List,
+) ([]*backend.BlockMeta, []*backend.CompactedBlockMeta, error) {
+	span, derivedCtx := opentracing.StartSpanFromContext(ctx, "Poller.pollTenantBlocks")
+	defer span.Finish()
+
+	currentBlockIDs, currentCompactedBlockIDs, err := p.reader.Blocks(derivedCtx, tenantID)
 	if err != nil {
-		metricBlocklistErrors.WithLabelValues(tenantID).Inc()
-		return []*backend.BlockMeta{}, []*backend.CompactedBlockMeta{}, err
+		return nil, nil, err
 	}
 
-	bg := boundedwaitgroup.New(p.cfg.PollConcurrency)
-	chMeta := make(chan *backend.BlockMeta, len(blockIDs))
-	chCompactedMeta := make(chan *backend.CompactedBlockMeta, len(blockIDs))
+	readPreviousSpan, _ := opentracing.StartSpanFromContext(derivedCtx, "readPreviousResults")
```

Comment:
an event may be more appropriate here. same for the other spans below (assemble/learn/etc)

Reply:
Yep, I started with events, but since we don't have support for them in traceql I opted for using full spans.

Reply:
Sounds like a good incentive to get it in :)

Reply:
Agreed, but still trying to get this one in :)